### PR TITLE
fix(load): fix mysqld crash when loaded data column contains double enclosed char. (#1263)

### LINKED ISSUE #1263: bug: The instance crashed after I execute a statement that includes a condition to load data.
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

```sql
CREATE TABLE AD_PINSTANCE_LOG
   (ID bigint, 
	AD_CLIENT_ID bigint, 
	AD_ORG_ID bigint, 
	AD_PINSTANCE_ID bigint, 
	P_DATE DATETIME DEFAULT null, 
	P_MSG VARCHAR(255), 
	OWNERID bigint, 
	MODIFIERID bigint, 
	CREATIONDATE DATETIME, 
	MODIFIEDDATE DATETIME, 
	ISACTIVE CHAR(1) DEFAULT 'Y' NOT NULL, 
	 PRIMARY KEY (ID));

load data infile '/tmp/AD_PINSTANCE_LOG' into table AD_PINSTANCE_LOG
FIELDS TERMINATED BY ',' 
ENCLOSED BY '"' 
LINES TERMINATED BY '\n'
(
@ID,
@AD_CLIENT_ID,
@AD_ORG_ID,
@AD_PINSTANCE_ID,
@P_DATE,
@P_MSG,
@OWNERID,
@MODIFIERID,
@CREATIONDATE,
@MODIFIEDDATE,
@ISACTIVE
)
set
ID=NULLif(@ID,''),
AD_CLIENT_ID=NULLif(@AD_CLIENT_ID,''),
AD_ORG_ID=NULLif(@AD_ORG_ID,''),
AD_PINSTANCE_ID=NULLif(@AD_PINSTANCE_ID,''),
P_DATE=NULLif(@P_DATE,''),
P_MSG=NULLif(@P_MSG,''),
OWNERID=NULLif(@OWNERID,''),
MODIFIERID=NULLif(@MODIFIERID,''),
CREATIONDATE=NULLif(@CREATIONDATE,''),
MODIFIEDDATE=NULLif(@MODIFIEDDATE,''),
ISACTIVE=NULLif(@ISACTIVE,'')
;
ERROR 2013 (HY000): Lost connection to MySQL server during query
```

### Expected behavior

_No response_

### How To Reproduce

_No response_

### Environment

build information as follow:
        Repository address: https://github.com/stoneatom/stonedb.git:stonedb-5.7-dev
        Branch name: stonedb-5.7-dev
        Last commit ID: 29e3f7c
        Last commit time: Date:   Thu Jan 12 17:58:24 2023 +0800
        Build time: Date: Mon Jan 16 08:21:32 CST 2023

### Are you interested in submitting a PR to solve the problem?

- [X] Yes, I will!

### PR DESCRIPTION
**Pull request by lujiashun**
## Summary about this PR

Issue Number: close #1263

1. crash reason is the loader column number each time is not 65535 but report 65535;
    occurs when using limit clause or duplicate key;
3. support double enclosed char, ref:https://dev.mysql.com/doc/refman/5.7/en/load-data.html

## Tests Check List

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [X] New Feature
- [X] Bug Fix
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/tianmu/loader/parsing_strategy.cpp [hunk 1]
```diff
@@ -113,15 +113,23 @@ inline bool ParsingStrategy::SearchUnescapedPattern(const char *&ptr, const char
...
+      } else
         ++ptr;
     }
   } else if (size == 2) {
     --search_end;
     while (ptr < search_end && (*ptr != *c_pattern || ptr[1] != c_pattern[1])) {
       if (escape_char_ && *ptr == escape_char_)
         ptr += 2;
-      else
+      else if (string_qualifier_ && *ptr == string_qualifier_ && ptr + 1 < search_end &&
```
Comment (wisehead) [inline storage/tianmu/loader/parsing_strategy.cpp]: this piece of code is repeated 4 times, how about add a new function for it?
----
(string_qualifier_ && *ptr == string_qualifier_ && ptr + 1 < search_end &&
               *(ptr + 1) == string_qualifier_)

File: mysql-test/suite/tianmu/std_data/issue1263-2.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+"765893","37","27","44221","2021-06-28 02:01:16","计算售罄率出错：ORA-04036: PGA memory used by the instance exceeds PGA_AGGREGATE_LIMITORA-06512: at ""BOSNDS3.O2O_STORESALERATE_SET"", line 49
```
Comment (RingsC) [inline mysql-test/suite/tianmu/std_data/issue1263-2.txt:1]: what will happens?  """ xxx """, and """" xxxx """"

File: storage/tianmu/loader/parsing_strategy.cpp (hunk 1 above)
Comment (lujiashun) [reply storage/tianmu/loader/parsing_strategy.cpp]: fixed it.

File: mysql-test/suite/tianmu/std_data/issue1263-2.txt (hunk 2 above)
Comment (lujiashun) [reply mysql-test/suite/tianmu/std_data/issue1263-2.txt:1]: add these test case . Please check the MTR result file or datafile(issue1263-2.txt and issue1263-3.txt). 
1. double enclosed char  will be interpred as a normal char;
2. singe enclosed char before field-terminated string will be interpred as enclose char
3. sing enclosed char with non-filed-termniated string will be interpreted as a normal char;

**Review by wisehead — APPROVED** [review]

File: mysql-test/suite/tianmu/std_data/issue1263-2.txt (hunk 2 above)
Comment (lujiashun) [reply mysql-test/suite/tianmu/std_data/issue1263-2.txt:1]: PTAL, @RingsC

**Review by DandreChen — APPROVED** [review]
LGTM

**Pull request merged**